+nix/Utils.m

```diff
@@ -157,13 +157,19 @@
                 error(err);
             end
 
+            if (maxDepth < 0)
+                err.identifier = 'NIXMX:InvalidArgument';
+                err.message = 'Search depth must be positive integers or logicals.';
+                error(err);
+            end
+
             nix.Utils.validFilter(filter, val);
 
             % transform matlab to c++ style index
-            md = nix.Utils.handleIndex(maxDepth);
+            %md = nix.Utils.handleIndex(maxDepth);
```

Comment:
Just kill this? We do have revision control to keep the history.

Reply:
Can do.

Reply:
Can do